src/components/AssetsList/index.jsx

```diff
@@ -0,0 +1,221 @@
+    });
+
+    this.props.selectAsset(asset);
+  }
+
+  getSelectedAssetIndex = (assetsList, selectedAsset) => (assetsList.findIndex(
+    asset => asset.id === selectedAsset.id));
+
+  getImageThumbnail(thumbnail) {
+    const baseUrl = this.props.courseDetails.base_url || '';
+
+    return (
+      <div className={styles['assets-list-image-preview-container']}>
+        {
+          thumbnail ?
+            <img className={styles['assets-list-image-preview-image']} src={`${baseUrl}${thumbnail}`} alt="" data-identifier="asset-image-thumbnail" /> :
+            <WrappedMessage message={messages.assetsListNoPreview} >
+              {displayText => <span className={classNames('text-center')} data-identifier="asset-image-thumbnail">{displayText}</span>}
+            </WrappedMessage>
+        }
+      </div>
+    );
+  }
+
+  getAssetsListHeader = () => (
+    <div className={classNames(styles['list-header'], 'row')}>
+      <span className="col-6 offset-3" data-identifier="asset-file-name" id="name-header"><WrappedMessage message={messages.assetsListNameLabel} /></span>
+      <span className="col-3" data-identifier="asset-date-added" id="date-added-header"><WrappedMessage message={messages.assetsListDateLabel} /></span>
+    </div>
+  )
+
+  getThumbnailElement = thumbnail => (
+    <span aria-hidden className="col">{this.getImageThumbnail(thumbnail)}</span>
+  );
+
+  getDisplayNameElement = (displayName, index) => (
+    <span id={`asset-name-${index}`} className="col-6" data-identifier="asset-file-name">{displayName}</span>
+  );
+
+  getDateAddedElement = (dateAdded, index) => (
+    <span id={`asset-date-${index}`} className="col-3" data-identifier="asset-date-added">{dateAdded}</span>
+  );
+
+  getAssetListItem = (asset, index) => {
+    const isSelected = this.props.selectedAsset.id === asset.id;
+
+    return (<li
+      // prevent clicking on list item from triggering parent's focus
+      onMouseDown={(e) => { e.preventDefault(); this.onAssetClick(asset, index); }}
+      className={classNames('list-group-item list-group-item-action', { active: isSelected })}
+      id={`asset-list-option-${index}`}
+      aria-selected={isSelected}
+      role="option"
+      key={asset.display_name}
+      tabIndex="-1"
+      aria-labelledby={`name-header asset-name-${index} date-added-header asset-date-${index}`}
+    >
+      <div className="row">
+        {this.getThumbnailElement(asset.thumbnail)}
+        {this.getDisplayNameElement(asset.display_name, index)}
+        {this.getDateAddedElement(asset.date_added, index)}
+      </div>
+    </li >);
+  }
+
+  handleKeyDown = (e) => {
+    switch (e.key) {
+      case 'ArrowDown': {
+        // prevent scrolling entire modal body with arrow keys
+        e.preventDefault();
+
+        if (this.state.selectedAssetIndex < this.props.assetsListMetadata.pageSize - 1) {
+          this.setState(
+            { selectedAssetIndex: this.state.selectedAssetIndex + 1,
```

Comment:
Because this `setState` relies on existing state I think we need to use the [functional `setState`](https://medium.freecodecamp.org/functional-setstate-is-the-future-of-react-374f30401b6b).

```jsx
this.setState((state, props) => ({
    selectedAssetIndex: state.selectedAssetIndex + 1,
    selectedAssetPage: props.assetListMetadata.page,
}, () => {
    this.props.selectAsset(this.props.assetsList[this.state.selectedAssetIndex]);
}));
```